### PR TITLE
Save Mode

### PR DESCRIPTION
**Pull request by marsishandsome**
TiSpark only support `Append` SaveMode.

`Append` in TiSpark means `upsert`,
1. if primary key exists in db, data will be updated
2. if no same primary key exists, data will be inserted.

| SaveMode | Support | Semantics |
| -------- | ------- | --------- |
| Append | true | TiSpark's `Append` means upsert. If primary key is same, data will be updated; if no same primary key exists, data will be inserted.  |
| Overwrite | false |  - |
| ErrorIfExists | false | - |
| Ignore | false | - |

Currently TiSpark only support writing data to such tables:
1. the table does not contain a primary key
2. the table's primary key is `TINYINT`、`SMALLINT`、`MEDIUMINT` or `INTEGER`
3. the table's primary key is `auto increment`

### CODE REVIEW

**Comment by marsishandsome** [comment]
@birdstorm @zhexuany PTAL

File: tikv-client/src/main/java/com/pingcap/tikv/codec/TableCodec.java [hunk 1]
```diff
@@ -24,6 +21,9 @@
               colTypes.length, colIDs.size()));
     }
 
+    // Row layout: colID1, value1, colID2, value2, .....
```
Comment (zhexuany) [inline tikv-client/src/main/java/com/pingcap/tikv/codec/TableCodec.java]: this comment belongs to `encodeRow`

Comment (marsishandsome) [reply tikv-client/src/main/java/com/pingcap/tikv/codec/TableCodec.java]: fix

**Pull request merged**